### ISSUE TITLE
Simplify the calling interface for `eigs`, `eighs`, and `svds`

### ISSUE THREAD

**Issue by loiseaujc** [nekStab/LightKrylov#52]
At the moment, the calling interface for `eigs` is

```
eigs(A, X, eigvecs, eigvals, residuals, info, nev, tolerance, verbosity, transpose)
```

Similar interfaces are available for `eighs` and `svds`. This was adapted from the original api of `nekStab` which dates back to 2011-ish. Carrying around the Krylov basis `X` and the components `eigvecs` of the eigenvectors within this subspace is very convenient. The primary reason was that we do no want to extend the `abstract vector` to complex-valued vectors.

Note however that, by default, LAPACK `dgeev` returns the complex eigenvectors as a real-valued array with the odd-columns containing the real part, and the even ones the imaginary part. In `utils.deig` we actually reconstruct the complex-valued eigenvectors and re-order the eigenvalues. How about we skip this last bit and have `eigs` return the real/imaginary parts of the high-dimensional eigenvectors directly ?

Basically, what I have in mind is, using the function @Simkern is writing (`get_mat`), we actually return `X = X @ eigvecs` (where `eigvecs` is the real-valued output of `dgeev`). The calling interface would reduce to

```
eigs(A, X, eigvals, ...)
```

where `X` in `intent(inout)`. As the computation proceeds, `X` stores the basis for the Krylov subspace. Once we have converged the desired number of eigenvalues, we return in the same `X` the actual real and imaginary parts of the eigenvectors. I think not having to carry around `eigvecs` would be easier for everyone. It is a breaking change in the API though and we would need to make the appropriate changes in `nekStab`. What do you think @ricardofrantz ?

PS : Obviously, the same would apply to `eighs` and `svds`.
